internal/controllers/hostedclusters/adapter_hostedcluster.go

```diff
@@ -0,0 +1,38 @@
+package hostedclusters
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type hostedCluster interface {
+	ClientObject() client.Object
+}
```

Comment:
>we only need that for other controllers that have a Cluster- and non Cluster- struct.
Ohh yeah, that is why I didn't put the `generic` prefix. 

I do think it would be nice to wrap the unstructured object. Are you saying that should be done in `hostedcluster_controller.go`?

Reply:
Wrapping the unstructured object makes sense :)
But then this is not an `adapter_`

Reply:
Ahhhh :D Okay thanks!